tests/integration/features/backup_multiple_disks_replicated.feature

```diff
@@ -279,3 +279,34 @@ Feature: Backup & Restore multiple disks and S3 with replication
     Examples:
     | number |
     | 8      |
+
+Scenario: Inplace data restore.
+  Given we use the same object storage bucket for clickhouse01 as on clickhouse02
+  And ch-backup configuration on clickhouse02
+  """
+    restore:
+      use_inplace_cloud_restore: True
+  """
+  
+  And we have executed queries on clickhouse01
+  """
+  CREATE DATABASE IF NOT EXISTS test_db;
+
+  CREATE TABLE test_db.table_01 (CounterID UInt32, UserID UInt32)
+  ENGINE = MergeTree() ORDER BY UserID SETTINGS storage_policy = 's3';
+
+  CREATE TABLE test_db.table_02 (CounterID UInt32, UserID UInt32)
+  ENGINE = ReplicatedMergeTree('/clickhouse/tables/shard_01/test_db.table_02', '{replica}')
+  ORDER BY UserID
+  SETTINGS storage_policy = 's3';
+
+  INSERT INTO test_db.table_01 SELECT number%30, number FROM system.numbers LIMIT 1000;
+  INSERT INTO test_db.table_02 SELECT number%30, number FROM system.numbers LIMIT 1000;
+  """
+  When we create clickhouse01 clickhouse backup
+  When we restore clickhouse backup #0 to clickhouse02
```

Comment:
Suggest to use the same host for testing in-place restore. Otherwise it's not in-place :)

And it would be great to check that restore is actually in-place. For example, check object count in S3 or check that references to S3 objects in local metadata of S3 disks after restore are the same as at backup creation time.